### PR TITLE
I needed these two changes in order to "make cold" on FreeBSD.

### PR DESCRIPTION
**Pull request by zhenya1007**

### CODE REVIEW

File: src_ext/Makefile [hunk 1]
```diff
@@ -70,7 +70,7 @@ endef
 
 %.stamp: %.download
 	mkdir -p tmp
-	cd tmp && $(if $(patsubst %.tar.gz,,$(URL_$*)),bunzip2,gunzip) -c ../$(notdir $(URL_$*)) | tar x
+	cd tmp && tar x$(if $(patsubst %.tar.gz,,$(URL_$*)),j,z)f ../$(notdir $(URL_$*))
```
Comment (dra27) [inline src_ext/Makefile]: This is a revert of e740dd6b8880462c5b581a6b404917451611cb47 and can't go through as-is: see https://github.com/ocaml/opam/issues/2158

I'm being lazy and not booting my FreeBSD VM, but is it really possible to have `tar -j` working yet no `bunzip2`? Would it be better to use `-d` with the `bzip2` or `gzip` commands instead?

**Comment by zhenya1007** [comment]
I think the problem is `tar x`.  The man page says,

>      -f file, --file file
>           Read the archive from or write the archive to the specified file.
>            The filename can be - for standard input or standard output.  The
>           default varies by system; on FreeBSD, the default is */dev/sa0*; on
>            Linux, the default is /dev/st0.

I saw `tar` complain that it cannot read from `/dev/sa0`.  

Let me try `tar xf -`: with any luck that works for everyone.

**Comment by zhenya1007** [comment]
Yup, `tar xf -` works on FreeBSD.  The [Solaris man page I found](http://docs.oracle.com/cd/E19109-01/tsolaris8/817-0879/6mgl9vnhn/index.html) suggests that it should work on Solaris, also.  

With any luck, this change can now be merged.  If you need me to squash the commits in my branch, please let me know.

**Comment by AltGr** [comment]
Thanks!

**Pull request merged**